pkg/provisioningv2/rke2/planner/store.go

```diff
@@ -160,28 +160,36 @@ func SecretToNode(secret *corev1.Secret) (*plan.Node, error) {
 	}
 	planData := secret.Data["plan"]
 	appliedPlanData := secret.Data["appliedPlan"]
+	appliedChecksum := string(secret.Data["applied-checksum"])
+	failedChecksum := string(secret.Data["failed-checksum"])
 	output := secret.Data["applied-output"]
 	appliedPeriodicOutput := secret.Data["applied-periodic-output"]
 	probes := secret.Data["probe-statuses"]
 	failureCount := secret.Data["failure-count"]
-	maxFailures := secret.Data["max-failures"]
+	maxFailures := secret.Data["max-failures"] // max-failures is set by Rancher, if it is not set, it will default to -1 (on the system-agent side) and attempt to apply forever
 
-	if len(failureCount) > 0 && len(maxFailures) > 0 {
+	planApplied := bytes.Equal(planData, appliedPlanData)
+
+	result.Failed = false
```

Comment:
Not needed. A `bool` will be `false` on struct initialization.

Reply:
removed